cprover_bindings/src/goto_program/typ.rs

```diff
@@ -104,6 +146,12 @@ pub struct Parameter {
     base_name: Option<InternedString>,
 }
 
+/// Type for function contracts, loop contracts, etc.
+#[derive(Debug, Clone)]
+pub enum Contract {
+    FunctionContract { ensures: Vec<Expr>, requires: Vec<Expr>, variables: Vec<Expr> },
```

Comment:
Can you add a comment on what `variables` is?

Reply:
Added comment. Moved contract to its own module.